app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @new_comment = params.require(:comment).permit(:text)
+    @post = Post.find(params.require(:post_id))
+    @comment = Comment.new(author: current_user, post: @post, text: @new_comment[:text])
+    if @comment.save
+      flash[:success] = 'Comment created succefully!'
+      redirect_to user_post_path(@post, id: params.require(:post_id))
+    else
+      flash.now[:error] = "The comment couldn't be created!"
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] I would like to recommend you add the params method below :arrow_down: you can check the following [link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) to know more about it :+1:

Reply:
Thank you for your feedback, due to the deadline I am skipping the optional tasks but I will surely act upon them on coming Sunday.